rpcs3/Emu/Cell/SPURecompiler.cpp

```diff
@@ -8275,9 +8321,34 @@ class spu_llvm_recompiler : public spu_recompiler_base, public cpu_translator
 	{
 		// TODO
 		if (g_cfg.core.spu_accurate_xfloat)
+		{
 			set_vr(op.rt, get_vr<f64[4]>(op.rb));
+			// const auto [a, b] = get_vrs<f64[4]>(op.ra, op.rb);
+
+			// const auto mask_se = splat<s64[4]>(0xfff0000000000000ull);
+			// const auto mask_bf = splat<s64[4]>(0x000fff8000000000ull);
+			// const auto mask_sf = splat<s64[4]>(0x0000007fe0000000ull);
+			// const auto mask_yf = splat<s64[4]>(0x0000ffffe0000000ull);
+
+			// const auto base = bitcast<f64[4]>((bitcast<s64[4]>(b) & mask_bf) | 0x3ff0000000000000ull);
+			// const auto step = fpcast<f64[4]>(bitcast<s64[4]>(b) & mask_sf) * fsplat<f64[4]>(std::exp2(-13.f));
+			// const auto yval = fpcast<f64[4]>(bitcast<s64[4]>(a) & mask_yf) * fsplat<f64[4]>(std::exp2(-19.f));
+			// set_vr(op.rt, bitcast<f64[4]>((bitcast<s64[4]>(b) & mask_se) | (bitcast<s64[4]>(base - step * yval) & ~mask_se)));
+		}
 		else
-			set_vr(op.rt, get_vr<f32[4]>(op.rb));
+		{
+			const auto [a, b] = get_vrs<s32[4]>(op.ra, op.rb);
```

Comment:
Thing is with FI instruction is that it "fixes" the result of FRSQEST on realhw. Here FRSQEST does exactly what the combination of FRSQEST+FI does so there is no need in FI's fixup. This patern can be detected and used to eliminate FI in this case.